src/libraries/Microsoft.Extensions.Primitives/tests/CompositeChangeTokenTest.cs

```diff
@@ -148,5 +148,66 @@ public async Task RegisteredCallbackGetsInvokedExactlyOnce_WhenMultipleConcurren
             // Assert
             Assert.Equal(1, count);
         }
+        
+        [Fact]
+        public void ShouldNotCollectDisposablesIfChangedTokenEncountered()
+        {
+            // Arrange
+            var firstCancellationTokenSource = new CancellationTokenSource();
+            var secondCancellationTokenSource = new CancellationTokenSource();
+            var thirdCancellationTokenSource = new CancellationTokenSource();
+            var count = 0;
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> {
+                new ProxyCancellationChangeToken(firstCancellationTokenSource.Token, disposing: () => count++),
+                new ProxyCancellationChangeToken(secondCancellationTokenSource.Token, disposing: () => count++),
+                new ProxyCancellationChangeToken(thirdCancellationTokenSource.Token, disposing: () => count++) });
+
+            // Act
+            firstCancellationTokenSource.Cancel();
+            compositeChangeToken.RegisterChangeCallback(_ => { }, null);
+            secondCancellationTokenSource.Cancel();
+
+            // Assert
+            Assert.Equal(1, count);
+        }
+    }
+
+    internal class ProxyCancellationChangeToken : IChangeToken
+    {
+        private readonly CancellationChangeToken _cancellationChangeToken;
+        private readonly Action _disposing;
+
+        public ProxyCancellationChangeToken(CancellationToken cancellationToken, Action disposing)
+        {
+            _cancellationChangeToken = new(cancellationToken);
```

Comment:
```suggestion
            _cancellationChangeToken = new CancellationChangeToken(cancellationToken);
```

Just like `var` without the right-hand side being obvious on the Type, we don't use `new()` without the left-hand side being obvious on the Type.